### PR TITLE
BEMHTML: fix rendering mixes for namesake elems

### LINKED ISSUE #218: Mix mode is little bit broken since v5.1
http://goo.gl/GdhxL8

BH doesn't have tests for BEMHTML modes so we need to write our own.

### PR DESCRIPTION
**Pull request by zxqfox**
Fixes #218

### CODE REVIEW

File: lib/bemhtml/index.js [hunk 1]
```diff
@@ -216,7 +216,8 @@ BEMHTML.prototype.renderMix = function renderMix(entity,
     var hasItem = false;
 
     if (item.elem) {
-      hasItem = item.elem !== entity.elem && item.elem !== context.elem;
+      hasItem = item.elem !== entity.elem && item.elem !== context.elem ||
+        item.block && item.block !== entity.block;
```
Comment (miripiruni) [inline lib/bemhtml/index.js]: Oh, my…

Thank you, @zxqfox

**Comment by miripiruni** [comment]
LGTM.

Merge?

**Comment by zxqfox** [comment]
I'd say it's a patch, not a minor.

**Comment by miripiruni** [comment]
@zxqfox OK, 6.0.1 and 5.1.1.

**Comment by zxqfox** [comment]
Do we ready to merge this? ^.^

**Comment by miripiruni** [comment]
Yes.

**Comment by veged** [comment]
LGTM

**Pull request merged**